Makefile

```diff
@@ -144,7 +145,7 @@ $(STM_C_INCLUDES)
 # compile gcc flags
 ASFLAGS = $(MCU) $(AS_DEFS) $(AS_INCLUDES) $(OPT) -Wall -fdata-sections -ffunction-sections
 
-CFLAGS += $(MCU) $(C_DEFS) $(C_INCLUDES) $(OPT) -Wall -fdata-sections -ffunction-sections -Werror=return-type
+CFLAGS += $(MCU) $(C_DEFS) $(C_INCLUDES) $(OPT) -Wall -fdata-sections -ffunction-sections -Werror=return-type -Werror=implicit-function-declaration -Werror=int-conversion
```

Comment:
## -Werror=implicit-function-declaration
Prevents the compiler from being okay with randomly used functions being used when not imported
  [Example](https://codeql.github.com/codeql-query-help/cpp/cpp-implicit-function-declaration/): 
```C
/* '#include <stdlib.h>' was forgotten */
          
int main(void) {
        /* 'int malloc()' assumed */
	unsigned char *p = malloc(100);
	*p = 'a';
	return 0;
}
```
## -Werror=int-conversion
Prevents the user from trying to convert a bigger data type like uint32_t to a smaller data type like uint8_t. This also applies when trying to assign a int to a int*. Can be overcome with an explicit cast.
```c
#include <stdio.h>

int main() {
    uint32_t largeNumber = 1000;
    uint8_t smallNumber;

    smallNumber = largeNumber; // Conversion from 32-bit to 8-bit will cause loss of data

    printf("Value of smallNumber: %d\n", smallNumber);

    return 0;
}
```